bolt-sidecar/src/lib.rs

```diff
@@ -2,6 +2,11 @@
 #![warn(missing_debug_implementations, missing_docs, rustdoc::all)]
 #![deny(unused_must_use, rust_2018_idioms)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+
+// Manually silence "rustls" crate being unused
+#[allow(unused_extern_crates)]
+extern crate rustls;
```

Comment:
Why is this needed?

Reply:
because rustls isn't used in the lib/ code, so the "unused crate" warning is triggered, but we do use it in bin/sidecar.rs, which isn't checked by this particular rule.

Reply:
Got it, related to #706